sdk/helper/ldaputil/client.go

```diff
@@ -388,6 +388,11 @@ func (c *Client) performLdapFilterGroupsSearchPaging(cfg *ConfigEntry, conn Pagi
 		return make([]*ldap.Entry, 0), nil
 	}
 
+	if *cfg.MaximumPageSize < 0 {
+		c.Logger.Warn("maximumpagesize is negative, will not query server")
+		return make([]*ldap.Entry, 0), nil
+	}
+
```

Comment:
I suggest not adding this code, as it is effectively unreachable - this function is only invoked when this condition is false.